### PR TITLE
Fix CA1725 in solution

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

**Comment by Youssef1313** [comment]
@mavasani Before moving further with fixing the build failures (due to naming conflicts), do you want to exclude public api surface? or it doesn't matter for this repository?

File: .editorconfig [hunk 1]
```diff
@@ -267,8 +267,7 @@ dotnet_diagnostic.CA1711.severity = suggestion
 dotnet_diagnostic.CA1309.severity = suggestion
 
 # CA1725: Parameter names should match base declaration
-# TODO: Enable this rule as warning and fix violations.
-dotnet_diagnostic.CA1725.severity = suggestion
+dotnet_diagnostic.CA1725.severity = warning
```
Comment (mavasani) [inline .editorconfig:270]: This setting is redundant. We set all CA rules to be warning by default here: https://github.com/dotnet/roslyn-analyzers/blob/edf95f72a20ed1c090a1639c8ec85bb5854fe09e/src/Directory.Build.props#L18-L22

File: src/Roslyn.Diagnostics.Analyzers/CSharp/CSharpSymbolDeclaredEventMustBeGeneratedForSourceSymbols.cs [hunk 2]
```diff
@@ -45,9 +45,9 @@ public CSharpCompilationAnalyzer(INamedTypeSymbol symbolType, INamedTypeSymbol c
 
             protected override ImmutableHashSet<string> SymbolTypesWithExpectedSymbolDeclaredEvent => s_symbolTypesWithExpectedSymbolDeclaredEvent;
 
-            protected override SyntaxNode? GetFirstArgumentOfInvocation(SyntaxNode node)
+            protected override SyntaxNode? GetFirstArgumentOfInvocation(SyntaxNode invocation)
             {
-                var invocation = (InvocationExpressionSyntax)node;
+                var invocation = (InvocationExpressionSyntax)invocation;
```
Comment (mavasani) [inline src/Roslyn.Diagnostics.Analyzers/CSharp/CSharpSymbolDeclaredEventMustBeGeneratedForSourceSymbols.cs:50]:
```suggestion
                var invocationExpression = (InvocationExpressionSyntax)invocation;
```

File: src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/CopyAnalysis/CopyAnalysis.CoreCopyAnalysisDataDomain.cs [hunk 3]
```diff
@@ -22,21 +22,21 @@ public CoreCopyAnalysisDataDomain(AbstractValueDomain<CopyAbstractValue> valueDo
                 _getDefaultCopyValue = getDefaultCopyValue;
             }
 
-            public override CoreCopyAnalysisData Merge(CoreCopyAnalysisData map1, CoreCopyAnalysisData map2)
+            public override CoreCopyAnalysisData Merge(CoreCopyAnalysisData value1, CoreCopyAnalysisData value2)
```
Comment (mavasani) [inline src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/CopyAnalysis/CopyAnalysis.CoreCopyAnalysisDataDomain.cs:25]: I'd rather just revert changes to this method and add a source suppression here.

File: src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysisContext.cs [hunk 4]
```diff
@@ -156,12 +156,12 @@ internal sealed class PropertySetAnalysisContext : AbstractDataFlowAnalysisConte
         public ImmutableDictionary<(INamedTypeSymbol, bool), string> HazardousUsageTypesToNames { get; }
 
 #pragma warning disable CA1307 // Specify StringComparison - string.GetHashCode(StringComparison) not available in all projects that reference this shared project
-        protected override void ComputeHashCodePartsSpecific(Action<int> addPart)
+        protected override void ComputeHashCodePartsSpecific(Action<int> builder)
```
Comment (mavasani) [inline src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysisContext.cs:159]: Honestly `addPart` seems a better name than `builder` for a delegate, but I don't have strong opinion either way.

File: src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/AnalysisEntityMapAbstractDomain.cs [hunk 5]
```diff
@@ -70,18 +70,18 @@ protected virtual void AssertValidAnalysisData(DictionaryAnalysisData<AnalysisEn
 #endif
         }
 
-        public override DictionaryAnalysisData<AnalysisEntity, TValue> Merge(DictionaryAnalysisData<AnalysisEntity, TValue> map1, DictionaryAnalysisData<AnalysisEntity, TValue> map2)
+        public override DictionaryAnalysisData<AnalysisEntity, TValue> Merge(DictionaryAnalysisData<AnalysisEntity, TValue> value1, DictionaryAnalysisData<AnalysisEntity, TValue> value2)
```
Comment (mavasani) [inline src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/AnalysisEntityMapAbstractDomain.cs:73]: Same here - would suggest reverting this method and just adding a source suppression.

**Review by mavasani — COMMENTED** [review]
Overall LGTM. Please fix the build errors + editorconfig and it should be good to merge.

File: src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/CopyAnalysis/CopyAnalysisContext.cs [hunk 6]
```diff
@@ -57,18 +57,18 @@ public sealed class CopyAnalysisContext : AbstractDataFlowAnalysisContext<CopyAn
...
             InterproceduralCopyAnalysisData? interproceduralAnalysisData)
         {
-            return new CopyAnalysisContext(ValueDomain, WellKnownTypeProvider, invokedControlFlowGraph, invokedMethod, AnalyzerOptions, InterproceduralAnalysisConfiguration,
+            return new CopyAnalysisContext(ValueDomain, WellKnownTypeProvider, invokedCfg, invokedMethod, AnalyzerOptions, InterproceduralAnalysisConfiguration,
                 PessimisticAnalysis, ExceptionPathsAnalysis, pointsToAnalysisResult, TryGetOrComputeAnalysisResult, ControlFlowGraph, interproceduralAnalysisData,
                 InterproceduralAnalysisPredicate);
         }
 
-        protected override void ComputeHashCodePartsSpecific(Action<int> addPart)
+        protected override void ComputeHashCodePartsSpecific(Action<int> builder)
```
Comment (sharwell) [inline src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/CopyAnalysis/CopyAnalysisContext.cs:71]: 💡 Consider renaming the base parameter in this case

File: src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAnalysis.PointsToDataFlowOperationVisitor.cs [hunk 7]
```diff
@@ -360,15 +360,15 @@ protected override void PostProcessArgument(IArgumentOperation operation, bool i
                 }
             }
 
-            protected override void ProcessReturnValue(IOperation? returnValue)
+            protected override void ProcessReturnValue(IOperation? returnValueOperation)
```
Comment (sharwell) [inline src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAnalysis.PointsToDataFlowOperationVisitor.cs:363]: 💡 Consider renaming the base parameter in this case

File: src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/AnalysisEntityMapAbstractDomain.cs (hunk 5 above)
Comment (sharwell) [reply src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/AnalysisEntityMapAbstractDomain.cs:73]: 💭 `value1` seems just as good as `map1` to me for this method

File: src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysisContext.cs (hunk 4 above)
Comment (Youssef1313) [reply src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysisContext.cs:159]: > Honestly `addPart` seems a better name than `builder` for a delegate, but I don't have strong opinion either way.

Will update in the name in base class to `addPart`.

File: src/Microsoft.CodeAnalysis.AnalyzerUtilities/PublicAPI.Unshipped.txt [hunk 8]
```diff
@@ -719,7 +719,7 @@ virtual Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.DataFlowOperationVisitor<TA
...
 virtual Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.DataFlowOperationVisitor<TAnalysisData, TAnalysisContext, TAnalysisResult, TAbstractAnalysisValue>.PostProcessArgument(Microsoft.CodeAnalysis.Operations.IArgumentOperation! operation, bool isEscaped) -> void
 virtual Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.DataFlowOperationVisitor<TAnalysisData, TAnalysisContext, TAnalysisResult, TAbstractAnalysisValue>.ProcessOutOfScopeLocalsAndFlowCaptures(System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.ILocalSymbol!>! locals, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.FlowAnalysis.CaptureId>! flowCaptures) -> void
-virtual Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.DataFlowOperationVisitor<TAnalysisData, TAnalysisContext, TAnalysisResult, TAbstractAnalysisValue>.ProcessReturnValue(Microsoft.CodeAnalysis.IOperation? returnValueOperation) -> void
+virtual Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.DataFlowOperationVisitor<TAnalysisData, TAnalysisContext, TAnalysisResult, TAbstractAnalysisValue>.ProcessReturnValue(Microsoft.CodeAnalysis.IOperation? returnValue) -> void
```
Comment (Youssef1313) [inline src/Microsoft.CodeAnalysis.AnalyzerUtilities/PublicAPI.Unshipped.txt:722]: This update is done manually. The RS diagnostic didn't show locally for me in Visual Studio.

**Review by mavasani — APPROVED** [review]

**Comment by mavasani** [comment]
@Youssef1313 I can merge once you fix the merge conflicts. Thanks.

**Pull request merged**